File: app/src/main/java/com/designpatterns/demo/behavioral/observer/java/Postman.java
package com.designpatterns.demo.behavioral.observer.java;

import java.util.ArrayList;
import java.util.List;


/**
 * 创建具体观察者
 *
 * @author zhangzhiguo
 * @version 1.0.0
 * @email: 810903987@qq.com
 * @date 2018/6/6 下午1:13
 * @project_name DesignPatternsDemo
 */
public class Postman implements Observable {

    private List<Observer> personList=new ArrayList<Observer>();

    @Override
    public void add(Observer observer) {
        personList.add(observer);
    }

    @Override
    public void remove(Observer observer) {
        personList.remove(observer);
    }

    @Override
    public void notify(String message) {
        for(Observer observer:personList){
            observer.update(message);
        }
    }
}

File: app/src/main/java/com/designpatterns/demo/behavioral/observer/kotlin/Postman.kt
package com.designpatterns.demo.behavioral.observer.kotlin


/**
 * 创建具体观察者
 *
 * @author  zhangzhiguo
 * @email: 810903987@qq.com
 * @date 2018/6/6 下午2:48
 * @project_name DesignPatternsDemo
 * @version 1.0.0
 *
 */
class Postman:Observable{

    val personList = ArrayList<Observer>()

    override fun add(observer: Observer) {
        personList.add(observer)
    }

    override fun remove(observer: Observer) {
        personList.remove(observer)
    }

    override fun notify(message: String) {
        for (observer in personList) {
            observer.update(message)
        }

    }


}